include/inexor/vulkan-renderer/io/byte_stream.hpp

```diff
@@ -0,0 +1,56 @@
+#pragma once
+
+#include "inexor/vulkan-renderer/world/cube.hpp"
+#include "inexor/vulkan-renderer/world/indentation.hpp"
+
+#include <array>
+#include <filesystem>
+#include <utility>
+#include <vector>
+
+namespace inexor::vulkan_renderer::io {
+class ByteStream {
+protected:
+    std::vector<std::uint8_t> m_buffer;
```

Comment:
Not sure if there's a way to make this `private` instead of `protected`. clang-tidy gives a warning here (as `protected` fields are generally frowned upon in c++).

Reply:
This member is be used by the ByteStreamWriter.

Reply:
Still, there's almost always a way to avoid `protected` members, but I think this can be fixed later, as it's not a serious issue.